pippo-controller/src/test/java/ControllerHandlerTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.junit.Test;
+import ro.pippo.controller.Controller;
+import ro.pippo.controller.DefaultControllerHandler;
+import ro.pippo.core.Param;
+import ro.pippo.core.ParamPattern;
+
+/**
+ * @author ScienJus
+ * @date 2016/3/2.
+ */
+public class ControllerHandlerTest {
+
+    @Test
+    public void testFindMethodWithDate() throws Exception {
```

Comment:
It's an idea to split this method in two methods (`testDateParam` and `testDateParamWithPattern`).
What we can do for both tests in the current implementation is to create a controller handler and to use an `assertNotNull(handler.getMethod())`.
I see a supplement assert related  to `parameterNames` if we create in `DefaultControllerHandler` a getter for `parameterNames` property. In this mode we can test the parameter names size and the paramater name.